pinecone/index_connection.go

```diff
@@ -49,11 +56,88 @@ func newIndexConnection(in newIndexParameters) (*IndexConnection, error) {
 	return &idx, nil
 }
 
+// Close closes the connection to a Pinecone index.
+//
+// Returns a pointer to an IndexConnection object and an error if the connection cannot be closed.
+//
+// Parameters:
+//   idx: The IndexConnection object.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//	  ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//	  SourceTag: "your_source_identifier", // optional
+//	}
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//	if err != nil {
+//		log.Fatalf("Failed to create Client: %v", err)
+//  }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//  if err != nil {
+//	  fmt.Println("Error:", err)
+//  }
+//
+//  idx, err := pc.Index(idxs[0].Host)
+//  if err != nil {
+//    fmt.Println("Error:", err)
+//	  return
+//	}
+//
+//  err = idx.Close()
+//	if err != nil {
+//	  fmt.Println("Error:", err)
```

Comment:
nit: These error messages could maybe be a bit more clear:
```suggestion
//    fmt.Println("Error connecting to index:", err)
//	  return
//	}
//
//  err = idx.Close()
//	if err != nil {
//	  fmt.Println("Error closing index connection:", err)
```